tests/drivers/adc/adc_api/src/test_adc.c

```diff
@@ -53,7 +53,9 @@
 static u16_t seq_buffer[BUFFER_SIZE];
 
 static struct adc_seq_entry entry = {
+#if defined(CONFIG_ADC_LEGACY_API)
 	.sampling_delay = 30,
```

Comment:
As this test does not support anything but the legacy API, have a check at the beginning of it:
#ifndef CONFIG_ADC_LEGACY_API
#error "Supports only legacy API"
#endif